syntax = "proto3";

service FileBasedQA {
  // 基于文本和文件的问答服务
  rpc QueryFile (QueryRequest) returns (QueryResponse);
}

message QueryRequest {
  string question = 1;  // 用户的问题
  bytes file_data = 2;  // 上传的文件内容（如二进制数据，支持PDF、文本等）
}

message QueryResponse {
  bool has_text = 1;         // 是否包含文本回答
  string text_answer = 2;    // 文本回答内容（可选）
  bool has_image = 3;        // 是否包含图像回答
  bytes image_data = 4;      // 图像的二进制数据（可选）
}
